src/stacktrace_windows-inl.h

```diff
@@ -27,33 +27,60 @@
 // (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
-// Portable implementation - just use glibc
+// Author: Andrew Schwartzmeyer
 //
-// Note:  The glibc implementation may cause a call to malloc.
-// This can cause a deadlock in HeapProfiler.
-#include <execinfo.h>
-#include <string.h>
+// Windows implementation - just use CaptureStackBackTrace
+
+#include "port.h"
 #include "stacktrace.h"
+#include "Dbghelp.h"
+#include <vector>
 
 _START_GOOGLE_NAMESPACE_
 
+static bool ready_to_run = false;
+class StackTraceInit {
+public:
+  HANDLE hProcess;
+  StackTraceInit() {
+    // Initialize the symbol handler
+    // https://msdn.microsoft.com/en-us/library/windows/desktop/ms680344(v=vs.85).aspx
+    hProcess = GetCurrentProcess();
+    SymSetOptions(SYMOPT_UNDNAME | SYMOPT_DEFERRED_LOADS);
+    SymInitialize(hProcess, NULL, true);
+    ready_to_run = true;
+  }
+  ~StackTraceInit() {
+    SymCleanup(hProcess);
+    ready_to_run = false;
+  }
+};
+
+static const StackTraceInit module_initializer;  // Force initialization
+
 // If you change this function, also change GetStackFrames below.
 int GetStackTrace(void** result, int max_depth, int skip_count) {
-  static const int kStackLength = 64;
-  void * stack[kStackLength];
-  int size;
-
-  size = backtrace(stack, kStackLength);
+  if (!ready_to_run) {
+    return 0;
+  }
   skip_count++;  // we want to skip the current frame as well
-  int result_count = size - skip_count;
-  if (result_count < 0)
-    result_count = 0;
-  if (result_count > max_depth)
-    result_count = max_depth;
-  for (int i = 0; i < result_count; i++)
-    result[i] = stack[i + skip_count];
+  if (max_depth > 64) {
+    max_depth = 64;
+  }
+  std::vector<void*> stack(max_depth);
```

Comment:
Note: On ELF systems, Symbolize and Demangle are meant to be async-signal-safe. As signal handler can be called even while malloc's internal lock is held, I think we use no malloc in Symbolize and Demangle. I don't know Windows signal handler. I assume it's OK to call any functions?

Hmm it seems to be no? https://msdn.microsoft.com/en-us/library/aa272921(v=vs.60).aspx

I'm also worried by Windows API calls to obtain stacktrace/symbol. I guess some of them are calling malloc internally.